lib/engine/game/g_1822_africa/game.rb

```diff
@@ -252,35 +335,47 @@ def setup_companies
             c.max_price = 10_000
           end
 
-          @companies = put_concession_to_front(@companies)
+          # Setup company abilities
+          @company_trains = {}
+          @company_trains['P1'] = find_and_remove_train_by_id('LP-0', buyable: false)
+          @company_trains['P2'] = find_and_remove_train_by_id('2P-0', buyable: false)
+          @company_trains['P3'] = find_and_remove_train_by_id('2P-1', buyable: false)
+          @company_trains['P4'] = find_and_remove_train_by_id('P+-0', buyable: false)
+          @company_trains['P5'] = find_and_remove_train_by_id('P+-1', buyable: false)
         end
 
-        def setup_bidboxes
-          # Set the owner to bank for the companies up for auction this stockround
-          bidbox_minors_refill!
-          bidbox_minors.each do |minor|
-            minor.owner = @bank
-          end
+        def reorder_companies_on_concession
+          next_concession = bank_companies.find { |c| concession?(c) }
+          next_concession_index = @companies.index(next_concession)
 
-          # Reset the choice for P9-M&GNR
-          @double_cash_choice = nil
-        end
+          # Only reorder if next concession is not the first company
+          return if bank_companies.index(next_concession).zero?
+
+          head = @companies[0...next_concession_index]
+          tail = @companies[next_concession_index..-1]
 
-        def put_concession_to_front(companies)
-          first_concession_index = companies.find_index { |c| c.id[0] == self.class::COMPANY_CONCESSION_PREFIX }
+          @companies = tail + head
 
-          head = companies[0...first_concession_index]
-          tail = companies[first_concession_index..-1]
+          unowned_head = head.filter { |c| !c.owner || c.owner == @bank }
```

Comment:
```suggestion
          unowned_head = head.select { |c| !c.owner || c.owner == @bank }
```

Reply:
I'm changing it though I don't quite get the difference, they look identical to me